### ISSUE TITLE
problems with AWS.CloudSearch.Domain.upload_documents

### ISSUE THREAD

**Issue by wyattisimo** [jkakar/aws-elixir#33]
I'm new to Elixir and attempting to upload document to AWS CloudSearch.

The `upload_documents` function is defined as follows:
```elixir
def upload_documents(client, input, options \\ []) do
  ...
end
```
If I do this:
```elixir
client = %AWS.Client{
  access_key_id: access_key_id,
  secret_access_key: secret_access_key,
  region: "us-east-1",
  endpoint: doc_endpoint
}

input = %{
  "documents" => [
    %{
      "type"   => "add",
      "id"     => "1",
      "fields" => %{
        "first_name"   => "Your",
        "last_name"    => "Mom",
        "phone_number" => "5551212"
      }
    }
  ]
}

AWS.CloudSearch.Domain.upload_documents(client, input)
```
Then I get this:
```
** (ArgumentError) argument error
    :erlang.list_to_integer([])
    src/hackney_url.erl:184: :hackney_url.parse_netloc/2
    src/hackney.erl:334: :hackney.request/5
    lib/httpoison/base.ex:396: HTTPoison.Base.request/9
    lib/aws/cloud_search_domain.ex:124: AWS.CloudSearch.Domain.perform_request/6
```

`options` defaults to `[]`, which seems to be the cause of the problem here. Is this a bug, or am I missing something?

Also, I'm not entirely certain what data should be sent for `options` anyway.

**Comment by jkakar** [comment]
Thanks for the report @wyattisimo.  `options` are HTTP options that are meant to be passed directly to `HTTPoison`.  I think you've uncovered a bug here.  I don't have time to look at it right now, but will dig in later this evening and see what I can learn.

**Comment by jkakar** [comment]
@wyattisimo I've spent awhile looking at this and it looks tricky to fix.  The clients in aws-elixir are all generated by a code generator... I'd never looked at Cloud Search until now, but I'm quickly discovering that it's full of special cases the code generator isn't able to handle easily.  At this point, I think I'm going to remove the client from aws-elixir because I don't have time to figure out how to support it.

I'm sorry that this is an unsatisfying outcome, but I'd rather be clear about what you can expect than not.  I do appreciate you bringing the issue up.

**Comment by jkakar** [comment]
I've just published v0.0.9 that removes support for Cloud Search to ensure that others don't encounter the same issues.

**Comment by wyattisimo** [comment]
😞
